packaging/os/yum_repository.py

```diff
@@ -0,0 +1,365 @@
+
+    def _load_dir(self, dirname):
+        for f in os.listdir(dirname):
+            if os.path.isfile(os.path.join(dirname, f)) and f.endswith('.repo'):
+                self._load_file(os.path.join(dirname, f))
+    
+    def load(self):
+        # Read repositories defined in main yum config
+        yum_conf = yum.config.StartupConf.config_file_path.default
+        self._load_file(yum_conf)
+        
+        # Read repositories from repo dirs
+        if len(yum.config.YumConf.reposdir.default) == 0:
+            raise YumRepositoryError("Failed to get repository directories from yum configuration")
+
+        self.repo_dirs = [ d for d in yum.config.YumConf.reposdir.default if os.path.isdir(d) ]
+        if len(self.repo_dirs) == 0:
+            # Appemp to create directory
+            os.makedirs(yum.config.YumConf.reposdir.default[0])
+            self.repo_dirs = [yum.config.YumConf.reposdir.default[0]]
+        else:
+            for d in self.repo_dirs:
+                self._load_dir(d)
+
+    def remove_repo(self, repo_id):
+        if self.has_key(repo_id):
+            if self.check_mode:
+                return True
+            repo_filename = self[repo_id]
+            cp = ConfigParser.ConfigParser()
+            cp.read(repo_filename)
+            if len(cp.sections()) == 1:
+                # Just removing the whole file
+                os.remove(repo_filename)
+            else:
+                cp.remove_section(repo_id)
+                cp.write(open(repo_filename, 'w'))
+            self.pop(repo_id)
+            return True
+        else:
+            return False
+
+    def put_repo(self, repo_id, params):
+        changed = False
+        cp = ConfigParser.ConfigParser()
+        if self.has_key(repo_id):
+            repo_filename = self[repo_id]
+            cp.read(repo_filename)
+            for p, v in params.items():
+                if p in [ i[0] for i in cp.items(repo_id) ]:
+                    if cp.get(repo_id, p) != v:
+                        cp.set(repo_id, p, v)
+                        changed = True
+                else:
+                    cp.set(repo_id, p, v)
+                    changed = True
+            
+            for p in cp.options(repo_id):
+                if not params.has_key(p):
+                    cp.remove_option(repo_id, p)
+                    changed = True
+
+        else:
+            # Creating new repo file
+            repo_filename = os.path.join(self.repo_dirs[0], repo_id + '.repo')
+            
+            if os.path.isfile(repo_filename):
+                cp.read(repo_filename)
+            
+            cp.add_section(repo_id)
+            for p, v in params.items():
+                cp.set(repo_id, p, v)
+            
+            changed = True
+
+        if changed and not self.check_mode:
+            cp.write(open(repo_filename, 'w'))
+        return changed
+
+
+                
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
```

Comment:
@binrush personally, I would add an new param called additional_attributes that takes a dict to allow for setting repo attributes for plugins.

I'd avoid disabling param checking just because it seems like it could create a confusing user experience.

An example showing what I'm thinking:
```
yum_repository: 
  id: example
  baseurl='http://example.com/repo/'
  additional_attributes:
    priority: 1
```